catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchSource.java

```diff
@@ -733,6 +737,10 @@ public ResourceResponse retrieveResource(URI uri, Map<String, Serializable> requ
     if (serializableId != null) {
       String metacardId = serializableId.toString();
       WebClient restClient = newRestClient(null, metacardId, true, null);
+      if (StringUtils.isNotBlank(username) && StringUtils.isNotBlank(password)) {
```

Comment:
❓ Do we need to account for the case if the password is blank? (E.g, a nonblank username with blank password)

Reply:
Is a blank password valid? If so, I can account for this.

Reply:
Hmm 🤔 . I don't think its illegal to be blank depending on what the client is connecting to. maybe @stustison might know?

Reply:
@rzwiefel The URLResourceReader checks for the username and password properties being null before it uses them https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L552. I think it should be safe to just remove this if statement which will allow blank username/passwords. I believe this should also fix the SonarQube Cognitive Complexity error. What do you think?